### PR TITLE
Adding support for terminology binding table

### PR DESCRIPTION
**Pull request by cerbs**
Adding terminology table for supported terminologies of "Example" binding strength

### CODE REVIEW

File: lib/resources/dstu2/allergy_intolerance.yaml [hunk 1]
```diff
@@ -109,6 +110,10 @@ fields:
                   {"system": "http://www.nlm.nih.gov/research/umls/rxnorm",
                    "code": "4125",
                    "display": "Ethiodized oil"}]}}
+  binding:
+    display: RxNorm
+    terminology: http://www.nlm.nih.gov/research/umls/rxnorm
```
Comment (daliboz) [inline lib/resources/dstu2/allergy_intolerance.yaml]: I believe we allow for snomed as well when the substance isn't a medication, and for negation.

**Comment by daliboz** [comment]
Is the goal to document all terminologies for existing resources, or just a subset and then follow up with each one? If it's all, we're missing some.

**Comment by cerbs** [comment]
A subset, specifically AllergyIntolerance and additionally DocumentReference because I'm familiar with it and it has actual known example values I could list to prove that functionality works. These changes were split off from the AllergyIntolerance Create/Update PR (https://github.com/cerner/fhir.cerner.com/pull/36) and is therefore focused there.

**Comment by daliboz** [comment]
+1

**Comment by MaxPhilips** [comment]
+1

**Pull request merged**